internal/services/ports.go

```diff
@@ -60,6 +63,34 @@ func getAvailablePort() (int, error) {
 	return 0, errors.New("no available port")
 }
 
+func selectPort(configPort int) (int, error) {
+	if configPort != 0 {
+		return configPort, nil
+	}
+
+	if portStr, exists := os.LookupEnv("PC_PORT_NUM"); exists {
```

Comment:
Thoughts on naming this `DEVBOX_PC_PORT_NUM`

Reasons:
* If user sets `PC_PORT_NUM` outside of devbox, it's going to make starting devbox services on multiple projects fail. Since this is the env name set by process compose itself, it's not unreasonable that people will set it.
* Keeping track of env variables is easier if we prefix them
* Avoids collisions